universal-application-tool-0.0.1/app/views/FileUploadViewStrategy.java

```diff
@@ -87,32 +96,6 @@ protected Tag renderQuestion(
     return Optional.of(button);
   }
 
-  /**
-   * Renders a form submit button for delete form if the file upload question is optional.
-   *
-   * <p>If an uploaded file is present, render the button text as delete. Otherwise, skip.
-   *
-   * <p>See {@link renderDeleteAndContinueFileUploadForms}.
-   */
-  Optional<Tag> maybeRenderSkipOrDeleteButton(Params params) {
-    if (hasAtLeastOneRequiredQuestion(params)) {
-      // If the file question is required, skip or delete is not allowed.
-      return Optional.empty();
-    }
-    String buttonText = params.messages().at(MessageKey.BUTTON_SKIP_FILEUPLOAD.getKeyName());
-    String buttonId = FILEUPLOAD_SKIP_BUTTON_ID;
-    if (hasUploadedFile(params)) {
-      buttonText = params.messages().at(MessageKey.BUTTON_DELETE_FILE.getKeyName());
-      buttonId = FILEUPLOAD_DELETE_BUTTON_ID;
-    }
-    Tag button =
-        submitButton(buttonText)
-            .attr(FORM, FILEUPLOAD_DELETE_FORM_ID)
-            .withClasses(ApplicantStyles.BUTTON_REVIEW)
-            .withId(buttonId);
-    return Optional.of(button);
-  }
-
```

Comment:
question (blocking): it looks like most of this is still the same between the two child classes, what prompted this change?

Reply:
A couple reasons: 1. We need to add the scripts for azure delete that aren't needed for aws, so I separated.
2. I enabled delete for even non-optional questions on Azure, but wasn't sure if it was something that I should do for AWS. I can pull this out into the parent class though if you'd prefer, just let me know